Installing showcase can flake

The showcase test "installs" showcase, which may fail to download things. Maybe needs a small retry loop?


```



ERROR error installing showcase: Output { status: ExitStatus(unix_wait_status(256)), stdout: "", stderr: "go: downloading github.com/googleapis/gapic-showcase v0.36.2\ngo: downloading cloud.google.com/go/iam v1.5.2\ngo: downloading cloud.google.com/go/longrunning v0.6.7\ngo: downloading github.com/golang/protobuf v1.5.4\ngo: downloading github.com/gorilla/mux v1.8.1\ngo: downloading github.com/googleapis/grpc-fallback-go v0.1.4\ngo: downloading github.com/soheilhy/cmux v0.1.5\ngo: downloading github.com/spf13/cobra v1.9.1\ngo: downloading github.com/spf13/viper v1.20.1\ngo: downloading golang.org/x/oauth2 v0.29.0\ngo: downloading cloud.google.com/go v0.120.0\ngo: downloading golang.org/x/sync v0.13.0\ngo: downloading google.golang.org/api v0.229.0\ngo: downloading google.golang.org/genproto v0.0.0-20250414145226-207652e42e2e\ngo: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20250414145226-207652e42e2e\ngo: downloading google.golang.org/grpc v1.71.1\ngo: downloading google.golang.org/protobuf v1.36.6\ngo: downloading github.com/google/uuid v1.6.0\ngo: downloading github.com/googleapis/gax-go/v2 v2.14.1\ngo: downloading google.golang.org/genproto/googleapis/api v0.0.0-20250414145226-207652e42e2e\ngo: downloading github.com/google/go-cmp v0.7.0\ngo: downloading golang.org/x/net v0.39.0\ngo: downloading github.com/spf13/pflag v1.0.6\ngo: downloading github.com/iancoleman/strcase v0.3.0\ngo: downloading golang.org/x/sys v0.32.0\ngo: downloading golang.org/x/text v0.24.0\ngo: downloading cloud.google.com/go/auth v0.16.0\ngo: downloading cloud.google.com/go/compute/metadata v0.6.0\ngo: downloading cloud.google.com/go/auth/oauth2adapt v0.2.8\ngo: downloading github.com/google/s2a-go v0.1.9\ngo: downloading go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.60.0\ngo: downloading golang.org/x/time v0.11.0\ngo: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.60.0\ngo: downloading github.com/googleapis/enterprise-certificate-proxy v0.3.6\ngo: downloading go.opentelemetry.io/otel v1.35.0\ngo: downloading go.opentelemetry.io/otel/metric v1.35.0\ngo: downloading go.opentelemetry.io/otel/trace v1.35.0\ngo: downloading github.com/felixge/httpsnoop v1.0.4\ngo: downloading github.com/go-logr/logr v1.4.2\ngo: downloading go.opentelemetry.io/auto/sdk v1.1.0\ngo: downloading github.com/go-logr/stdr v1.2.2\ngo: downloading golang.org/x/crypto v0.37.0\n/builder/home/go/pkg/mod/github.com/googleapis/gapic-showcase@v0.36.2/cmd/gapic-showcase/compliance_service.go:9:2: github.com/spf13/viper@v1.20.1: verifying module: github.com/spf13/viper@v1.20.1: reading https://sum.golang.org/tile/8/0/x139/854: 500 Internal Server Error\n" }
  |  
```
https://pantheon.corp.google.com/cloud-build/builds;region=us-central1/f3c99cac-c6a0-4a4a-822b-3f36223f9634?project=rust-sdk-testing&e=SqlStudioMysqlLaunch::SqlStudioMysqlEnabled&mods=monitoring_api_staging

--- a/tests/integration/src/showcase.rs
+++ b/tests/integration/src/showcase.rs
@@ -68,16 +68,15 @@ pub async fn run() -> Result<()> {
 
 async fn install() -> Result<String> {
     for backoff in [Some(5), Some(10), Some(20), None] {
-        match install_attempt().await {
+        let error = match install_attempt().await {
             Ok(path) => return Ok(path),
-            Err(e) => {
-                let Some(delay) = backoff else {
-                    return Err(e);
-                };
-                tracing::info!("install failed trying again in {backoff:?} seconds: {e:?}");
-                tokio::time::sleep(Duration::from_secs(delay)).await;
-            }
-        }
+            Err(e) => e,
+        };
+        let Some(delay) = backoff.map(Duration::from_secs) else {
+            return Err(error);
+        };
+        tracing::info!("install failed trying again in {backoff:?} seconds: {error:?}");
+        tokio::time::sleep(delay).await;
     }
     unreachable!("the retry loop should return");
 }

